### PR TITLE
Implement std::error::Error for errors

### LINKED ISSUE #247: PasrseError and error Trait
Hi,

I was wondering why the ParseError struct does not implement the [std::error::Error](https://doc.rust-lang.org/stable/std/error/trait.Error.html) trait? I was hoping to wrap it in a custom error enum type and use the "?" operator when calling parse functions. From what I've tried so far has not allowed me to do so.

### PR DESCRIPTION
**Pull request by xfix**
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-cssparser/256)

Fixes #247.

### CODE REVIEW

File: src/parser.rs [hunk 1]
```diff
@@ -62,6 +76,14 @@ pub struct BasicParseError<'i> {
     pub location: SourceLocation,
 }
 
+impl<'i> fmt::Display for BasicParseError<'i> {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "{}", self.kind.description())
+    }
+}
+
+impl<'i> Error for BasicParseError<'i> {}
```
Comment (nox) [inline src/parser.rs]: Why did you not implement `description` here?

Comment (xfix) [reply src/parser.rs]: The documentation for `Error` says: "This method is soft-deprecated. Although using it won’t cause compilation warning, new code should use Display instead and new impls can omit it."

**Comment by dutchmartin** [comment]
@SimonSapin @nox Could someone take another look at this PR?
If there are any issues with it, please let me know.
I am happy to fix them since I want to use this error feature in my own project.

File: src/parser.rs (hunk 1 above)
Comment (nox) [reply src/parser.rs]: Please implement it anyway, calling `BasicParseErrorKind::description`.

File: src/parser.rs [hunk 2]
```diff
@@ -62,6 +76,14 @@ pub struct BasicParseError<'i> {
     pub location: SourceLocation,
 }
 
+impl<'i> fmt::Display for BasicParseError<'i> {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "{}", self.kind.description())
```
Comment (nox) [inline src/parser.rs]: This won't print enough information, i.e. we end up not printing the payloads of `UnexpectedToken(_)` and `AtRuleInvalid(_)`.

File: src/parser.rs [hunk 3]
```diff
@@ -156,6 +178,17 @@ impl<'i, T> ParseError<'i, T> {
     }
 }
 
+impl<'i, T> fmt::Display for ParseError<'i, T> {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "{}", match &self.kind {
+            ParseErrorKind::Basic(basic_kind) => basic_kind.description(),
+            ParseErrorKind::Custom(_) => "Custom error",
+        })
```
Comment (nox) [inline src/parser.rs]: Please put the match outside and avoid `write!`.

```rust
match &self.kind {
    ParseErrorKind::Basic(basic) => basic.fmt(formatter),
    ParseErrorKind::Custom(_) => "custom error".fmt(formatter),
}
```

File: src/parser.rs [hunk 4]
```diff
@@ -156,6 +178,17 @@ impl<'i, T> ParseError<'i, T> {
     }
 }
 
+impl<'i, T> fmt::Display for ParseError<'i, T> {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "{}", match &self.kind {
+            ParseErrorKind::Basic(basic_kind) => basic_kind.description(),
+            ParseErrorKind::Custom(_) => "Custom error",
```
Comment (nox) [inline src/parser.rs]: Why is the payload disregarded here?

File: src/parser.rs [hunk 5]
```diff
@@ -156,6 +178,17 @@ impl<'i, T> ParseError<'i, T> {
...
+        })
+    }
+}
+
+impl<'i, T> Error for ParseError<'i, T> where T: fmt::Debug {}
```
Comment (nox) [inline src/parser.rs]: Implement `description` here.

**Pull request closed without merging**